libavcodec/qsvdec.c

```diff
@@ -497,9 +503,7 @@ static int qsv_decode(AVCodecContext *avctx, QSVContext *q,
         ++q->zero_consume_run;
         if (q->zero_consume_run > 1)
             ff_qsv_print_warning(avctx, ret, "A decode call did not consume any data");
-    } else if (!*sync && bs.DataOffset) {
-        ++q->buffered_count;
-    } else {
+    } else if (*sync) {
```

Comment:
why we need this if?

Reply:
In previous code, we enter this branch when sync is assigned by some value, but when I remove the second branch the logic will change so I add this judgement to keep the logic same.

Reply:
Well... It doesn't effect function, so removing it maybe better.